### ISSUE TITLE
train and test set in YOLO v3

### ISSUE THREAD

**Issue by jain2anki** [pjreddie/darknet#2120]
my dataset contains few images (~490 images for 2 classes). so it is fine if don't split the dataset into train and test set and use all the images for training? Does the size of test set affect the training of yolo?

**Comment by Rajasekhar06** [comment]
Nope, it does not affect training but while calculating the accuracy for your created model then you might need to provide the images which are never shown to the model to check how accurately it is getting predictions while calculating mAP other than that there is a no use of test and train set division

**Comment by jain2anki** [comment]
and how do we calculate accuracy in yolo? I just know how to detect the object, but is there some functionality to test the accuracy given a validation dataset?

**Comment by Rajasekhar06** [comment]
yes there is a functionality 
`./darknet detector map data/YourDataFile.data cfg/yourCfgFile.cfg backup/YourWeightsFile.weights -points 0 -thresh 0.1`
in data file you need to mention the path of your test set then it will go through all the images and annotations and compare them to the detected classes make sure these images are not from training set give completely new images to find the real accuracy.

**Comment by vishnuvardhan58** [comment]
Hello @Rajasekhar06 , I am training with 5 species and gave the command you mentioned above and I am getting the results shown in the below image , what exactly is the meaning of TP ,FP and FN for each class ? 
<img width="478" alt="all" src="https://user-images.githubusercontent.com/9161539/97790776-43c05300-1baa-11eb-85be-a3bfd64f3ad5.PNG">

**Comment by Fetulhak** [comment]
> Hello @Rajasekhar06 , I am training with 5 species and gave the command you mentioned above and I am getting the results shown in the below image , what exactly is the meaning of TP ,FP and FN for each class ?
> 
> <img alt="all" width="478" src="https://user-images.githubusercontent.com/9161539/97790776-43c05300-1baa-11eb-85be-a3bfd64f3ad5.PNG">

Tp is the true positive predictions: while FP is the false positives which are negative/background predicted as an object. Fn is the false-negative prediction, in this case, the predicted bbox has a lower IOU-thresh than the already setted value which is most of the time 0.5 but it can vary based on the problem at hand.

**Comment by JoshuaTern** [comment]
@Fetulhak , can you please explain to me what are all the parameter for? for my own training testing result
![finalDetectionResult](https://github.com/pjreddie/darknet/assets/134839989/7e2555cd-928f-4517-ab12-fe0e3ed01ec7)
